src/tools/rust-analyzer/crates/hir-ty/src/diagnostics/match_check/deconstruct_pat.rs

```diff
@@ -502,6 +504,7 @@ impl Constructor {
                 .iter()
                 .filter_map(|c| c.as_int_range())
                 .any(|other| range.is_covered_by(other)),
+            #[allow(unreachable_code)] // `Slice` is uninhabited, and stage1 emits a lint.
```

Comment:
It feels like maybe we should not look into `Adt`s for the unreachable code lint?

because the pattern of having the following feels really nice and i believe that these warnings are harmful here
```rust
pub(super) struct Slice {
    _unimplemented: Void,
}
```